testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/mock/BpmnExampleDataReader.java

```diff
@@ -0,0 +1,132 @@
+import java.io.ByteArrayInputStream;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import org.awaitility.core.ConditionTimeoutException;
+import org.camunda.bpm.model.xml.instance.ModelElementInstance;
+
+public class BpmnExampleDataReader {
+
+  private final CamundaClient client;
+  private final CamundaAssertAwaitBehavior awaitBehavior;
+
+  public BpmnExampleDataReader(
+      final CamundaClient client, final CamundaAssertAwaitBehavior awaitBehavior) {
+
+    this.client = client;
+    this.awaitBehavior = awaitBehavior;
+  }
+
+  public String readExampleData(final long processDefinitionKey, final String elementId) {
+    final String failureMessagePrefix =
+        String.format(
+            "BPMN Model [processDefinitionKey: '%s', elementId: '%s']",
+            processDefinitionKey, elementId);
+
+    return Optional.<ModelElementInstance>ofNullable(
+            buildModelInstance(failureMessagePrefix, processDefinitionKey)
+                .getModelElementById(elementId))
+        .map(parentElement -> queryParentElementForExampleData(failureMessagePrefix, parentElement))
+        .orElseThrow(
+            () -> {
+              final String failureMessage =
+                  String.format(
+                      "%s does not contain element with id '%s'", failureMessagePrefix, elementId);
+              return new BpmnExampleDataReaderException(failureMessage);
+            });
+  }
+
+  private BpmnModelInstance buildModelInstance(
+      final String failureMessagePrefix, final long processDefinitionKey) {
+
+    final AtomicReference<BpmnModelInstance> modelInstance = new AtomicReference<>();
+
+    try {
+      awaitBehavior.untilAsserted(
+          () -> {
+            final Optional<String> bpmnXml =
+                Optional.ofNullable(
+                    client.newProcessDefinitionGetXmlRequest(processDefinitionKey).send().join());
+
+            assertThat(bpmnXml).isPresent();
+
+            modelInstance.set(
+                Bpmn.readModelFromStream(new ByteArrayInputStream(bpmnXml.get().getBytes())));
+          });
+    } catch (final ConditionTimeoutException t) {
+      throw new BpmnExampleDataReaderException(
+          String.format("%s failed to parse the BPMN model.", failureMessagePrefix), t);
+    }
+
+    return modelInstance.get();
+  }
+
+  private String queryParentElementForExampleData(
+      final String failureMessagePrefix, final ModelElementInstance parentElement) {
+
+    return queryParentElementForExampleData(parentElement)
+        .orElseThrow(
+            () -> {
+              final String failureMessage =
+                  String.format(
+                      "%s has no example data. Example data must have the attribute name '%s' or "
+                          + "else it won't be recognized. Returning an empty JSON object.",
+                      failureMessagePrefix, ZeebeConstants.PROPERTY_EXAMPLE_DATA);
```

Comment:
🔧 We could shrink the failure message.

```suggestion
              final String failureMessage =
                  String.format(
                      "%s has no example data as a property with the name '%s'.",
                      failureMessagePrefix, ZeebeConstants.PROPERTY_EXAMPLE_DATA);
```